client/src/components/ProjectForm.js

```diff
@@ -255,10 +280,9 @@ export default function ProjectForm({
             </Grid>
             <Grid item xs="auto">
               <StyledButton
-                component={Link}
-                to="/projects"
                 variant="contained"
                 cursor="pointer"
+                onClick={handleOpen}
```

Comment:
The open/close logic needs a bit more work.

Right the modal is opens warning of changes even if no changes were made.

Modal should open if:
- Changes were made to the form the modal is referencing and
  - The user tries to navigate away from the current page
  - The user presses the Close button

The Modal should not open if:
- The user tries to navigate away from then page and no changes were made
- The user presses the close button and no changes were made

Some sort of `ModalShouldTrigger` prop might help so that the parent component can determine when the modal should mess with user navigation.

Reply:
@Spiteless  fixed